src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -62,9 +62,20 @@ public WebScriptHostManager(ScriptHostConfiguration config, ISecretManagerFactor
                 config.TraceWriter = systemTraceWriter;
             }
 
-            _secretManager = secretManagerFactory.Create(settingsManager, config.TraceWriter, new FileSystemSecretsRepository(webHostSettings.SecretsPath));
             _performanceManager = new HostPerformanceManager(settingsManager);
             _swaggerDocumentManager = new SwaggerDocumentManager(config);
+            
+            var secretStorageType = settingsManager.GetSetting("UseBlobSecretStorage"); //TODO: official parameter naming                        
+            if (secretStorageType != null) 
+            {
+                string storageString = AmbientConnectionStringProvider.Instance.GetConnectionString(ConnectionStringNames.Storage);
+                string siteHostId = settingsManager.AzureWebsiteDefaultSubdomain ?? config.HostConfig.HostId;
```

Comment:
Are you suggesting to use AzureWebsiteHostName instead of "settingsManager.AzureWebsiteDefaultSubdomain" or something different? AzureWebsiteDefaultSubdomain is based on AzureWebsiteHostName, but I preferred the simple friendly site name (like "sitea-staging") to the full AzureWebsiteHostname (like "sitea-staging.azurewebsites.net").